File: scripts/test_pdf.py
import sys
import fitz # PyMuPDF

pdf_document = "dokuments/Ürün Etiketleri/surup/FO KARAMEL SURUP-(sekerli).pdf"
doc = fitz.open(pdf_document)
text = ""
for page in doc:
    text += page.get_text()

with open("scripts/test_pdf_output.txt", "w", encoding="utf-8") as f:
    f.write(text)

doc.close()

File: scripts/test_pdf.mjs
import fs from 'fs';
import PDFParser from 'pdf2json';

const filePath = 'dokuments/Ürün Etiketleri/surup/FO KARAMEL SURUP-(sekerli).pdf';

async function test() {
    const pdfParser = new PDFParser(this, 1);

    pdfParser.on("pdfParser_dataError", errData => console.error(errData.parserError));
    pdfParser.on("pdfParser_dataReady", pdfData => {
        console.log("TEXT EXTRACTION SUCCESS:");
        console.log("------------------------");
        console.log(pdfParser.getRawTextContent());
    });

    pdfParser.loadPDF(filePath);
}

test();
